src/Altinn.App.Core/Infrastructure/Clients/Notifications/Email/EmailNotificationClient.cs

```diff
@@ -1,58 +1,45 @@
 using Altinn.App.Core.Configuration;
 using Altinn.App.Core.Internal.App;
-using Altinn.App.Core.Internal.Email;
-using Altinn.App.Core.Models.Email;
+using Altinn.App.Core.Internal.Notifications.Email;
+using Altinn.App.Core.Models.Notifications.Email;
 using Altinn.Common.AccessTokenClient.Services;
 using Microsoft.ApplicationInsights;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
-using Prometheus;
-using System.Diagnostics;
 using System.Net.Http.Headers;
 using System.Text.Json;
 
-namespace Altinn.App.Core.Infrastructure.Clients.Email;
-/// <summary>
-/// Implementation of the <see cref="IEmailNotificationClient"/> interface using a HttpClient to send
-/// requests to the Email Notification service.
-/// </summary>
-public sealed class EmailNotificationClient : IEmailNotificationClient
+namespace Altinn.App.Core.Infrastructure.Clients.Notifications.Email;
+
+internal sealed class EmailNotificationClient : IEmailNotificationClient
 {
+    private readonly ILogger<EmailNotificationClient> _logger;
     private readonly IHttpClientFactory _httpClientFactory;
     private readonly IAppMetadata _appMetadata;
     private readonly PlatformSettings _platformSettings;
     private readonly IAccessTokenGenerator _accessTokenGenerator;
-    private readonly TelemetryClient _telemetryClient;
-    private static readonly Counter _orderCount = Metrics
-        .CreateCounter("altinn_app_notification_order_request_count", "Number of notification order requests.", labelNames: ["type", "result"]);
+    private readonly TelemetryClient? _telemetryClient;
 
-    /// <summary>
-    /// Initializes a new instance of the <see cref="EmailNotificationClient"/> class.
-    /// </summary>
-    /// <param name="httpClientFactory"></param>
-    /// <param name="platformSettings">Api endpoints for platform services.</param>
-    /// <param name="appMetadata">The service providing appmetadata.</param>
-    /// <param name="accessTokenGenerator">An access token generator to create an access token.</param>
-    /// <param name="telemetryClient">Client used to track dependencies.</param>
     public EmailNotificationClient(
+        ILogger<EmailNotificationClient> logger,
         IHttpClientFactory httpClientFactory,
         IOptions<PlatformSettings> platformSettings,
         IAppMetadata appMetadata,
         IAccessTokenGenerator accessTokenGenerator,
-        TelemetryClient telemetryClient)
+        IServiceProvider serviceProvider)
```

Comment:
Would `TelemetryClient? telemetryClient = null` yield the same result? This would in my opinion make the constructor easier to read next time around, and a bit easier to use in tests.

Reply:
Yup! That works 👍